src/main/java/ru/sbt/mipt/oop/eventhandler/DoorEventHandler.java

```diff
@@ -0,0 +1,62 @@
+package ru.sbt.mipt.oop.eventhandler;
+
+import ru.sbt.mipt.oop.Event;
+import ru.sbt.mipt.oop.Room;
+import ru.sbt.mipt.oop.SensorCommand;
+import ru.sbt.mipt.oop.SmartHome;
+import ru.sbt.mipt.oop.commandsender.CommandSenderImpl;
+import ru.sbt.mipt.oop.door.Door;
+import ru.sbt.mipt.oop.light.Light;
+import ru.sbt.mipt.oop.type.CommandType;
+
+import static ru.sbt.mipt.oop.type.EventType.DOOR_CLOSED;
+import static ru.sbt.mipt.oop.type.EventType.DOOR_OPEN;
+
+public class DoorEventHandler implements EventHandler {
+	private final SmartHome smartHome;
+	private final CommandSenderImpl commandSender;
+
+	public DoorEventHandler(SmartHome smartHome) {
+		this.smartHome = smartHome;
+		this.commandSender = new CommandSenderImpl();
+	}
+
+	public void handle(Event event) {
+		for (Room room : smartHome.getRooms()) {
+			for (Door door : room.getDoors()) {
+				if (door.getId().equals(event.getObjectId())) {
+					if (event.getType() == DOOR_OPEN) {
+						doorOpen(room, door);
+					} else if (event.getType() == DOOR_CLOSED) {
+						doorClosed(room, door);
+					}
+				}
+			}
+		}
+	}
+
+	private void doorOpen(Room room, Door door) {
+		door.setOpen(true);
+		System.out.println("Door " + door.getId() + " in room " + room.getName() + " was opened.");
+	}
+
+	private void doorClosed(Room room, Door door) {
+		door.setOpen(false);
+		System.out.println("Door " + door.getId() + " in room " + room.getName() + " was closed.");
+		if (room.getName().equals("hall")) {
```

Comment:
https://github.com/iagoshkov/smart-home-2020/blob/master/STUDENT_GUIDE.md#%D1%80%D0%B0%D0%B7%D0%B4%D0%B5%D0%BB%D1%8F%D0%B9%D1%82%D0%B5-%D0%B8%D0%B7%D0%BC%D0%B5%D0%BD%D0%B5%D0%BD%D0%B8%D0%B5-%D0%BE%D0%B1%D1%8A%D0%B5%D0%BA%D1%82%D0%BE%D0%B2-%D0%B8-%D1%81%D1%86%D0%B5%D0%BD%D0%B0%D1%80%D0%B8%D0%B8-%D1%83%D0%BC%D0%BD%D0%BE%D0%B3%D0%BE-%D0%B4%D0%BE%D0%BC%D0%B0